### PR TITLE
LPD-28758: skip company creation for invalid admin emails

### PR DESCRIPTION
**Pull request by devopsmystic**

### CODE REVIEW

**Comment by devopsmystic** [comment]
ci:test:relevant

**Comment by devopsmystic** [comment]
Will work with @dantewang and @vicnate5 to get this reviewed

File: modules/apps/portal-instances/portal-instances-test/src/testIntegration/java/com/liferay/portal/instances/internal/configuration/test/PortalInstancesConfigurationFactoryTest.java [hunk 1]
```diff
@@ -93,10 +94,36 @@ public void testAddCompanyWithAllAdminProperties() throws Exception {
...
 			).build());
+
+		try {
+			_company = _companyLocalService.getCompanyByWebId(_webId);
+		}
+		catch (NoSuchCompanyException noSuchCompanyException) {
+			companyCreated = false;
+		}
+
+		Assert.assertFalse(companyCreated);
```
Comment (dantewang) [inline modules/apps/portal-instances/portal-instances-test/src/testIntegration/java/com/liferay/portal/instances/internal/configuration/test/PortalInstancesConfigurationFactoryTest.java:126]: @devopsmystic 

Firstly, for the current assertion, a better way is to use an `Assert.fail()` in the try block:
```
		try {
			_company = _companyLocalService.getCompanyByWebId(_webId);
			
			Assert.fail("reason");
		}
		catch (NoSuchCompanyException noSuchCompanyException) {
			// swallow it, or assert the message of the NSCE if SF complains about the exception not being handled
		}
```
With this, it fails immediately if the `getCompanyByWebId()` method doesn't throw any exception; no need to use a boolean variable.

Additionally, to make the assertion even more complete, use `LoggerTestUtil` to wrap whole logic and assert that the info log is correctly printed.

Comment (dantewang) [reply modules/apps/portal-instances/portal-instances-test/src/testIntegration/java/com/liferay/portal/instances/internal/configuration/test/PortalInstancesConfigurationFactoryTest.java:126]: By the way, please add ticket number to this commit.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@devopsmystic Please also watch https://github.com/liferay-core-infra/liferay-portal/pull/7454, your PR might need rebasing if that one gets merged first.

**Pull request closed without merging**